### ISSUE TITLE
String Constructors

### ISSUE THREAD

**Issue by staabm** [jquery/sizzle#175]
I read todays article from Mozilla regarding Strings:
https://hacks.mozilla.org/2012/12/performance-with-javascript-string-objects/

 tl;dr
do use String-Literals `var s = "xyz"` instead of String constructors `var s = new String("xyz")` whenever you can, because subsequent method calls or property lookups on a StringWrapped object (created by a String-constructor) are a lot slower in most cases.

So the next thing I thought about, maybe jQuery/Sizzle has such problematic cases, let's have a look. At the first look it was not obvious, but I think there Sizzle has one such a case inside current jQuery 1.8.3's code.

[Sizzle aliases](https://github.com/jquery/sizzle/blob/8d91687e64f137fac70a8f88d361c5bbb688299f/sizzle.js#L28) the `String` constructor to `Token` and afterwards [uses](https://github.com/jquery/sizzle/blob/8d91687e64f137fac70a8f88d361c5bbb688299f/sizzle.js#L1081) `new Token (..)`  to instantiate tokens.

Maybe my assumption is not correct or it is just micro, but I thought it's worth mentioning.
